### PR TITLE
create SubpathRemovalProcessor

### PR DESCRIPTION
**Pull request by BurnzZ**

### CODE REVIEW

File: duplicate_url_discarder/processors/subpath_removal.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            )
+
+        for arg in self.args:
+            if not isinstance(arg, int):
+                raise ValueError(
+                    f"{base_exception_msg} Encountered {type(arg)}: {arg}."
+                )
+
+    def process(self, input_url: str) -> str:
+        positions = [int(pos) for pos in self.args]
```
Comment (wRAR) [inline duplicate_url_discarder/processors/subpath_removal.py:31]: We already check that they are ints so is this needed?

File: duplicate_url_discarder/processors/subpath_removal.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from scrapy.utils.url import parse_url
+
+from .base import UrlProcessorBase
+
+
+class SubpathRemovalProcessor(UrlProcessorBase):
+    """Removes the i-th subpath (starts at 0) from the URL path if it exists.
+
+    An example would be removing the 0-th and 2nd subpath of a URL like
+    https://example.com/a/b/c/d would result in https://example.com/b/c.
```
Comment (Gallaecio) [inline duplicate_url_discarder/processors/subpath_removal.py:13]: From my understanding of the statement:

```suggestion
    https://example.com/a/b/c/d would result in https://example.com/b/d.
```

**Review by Gallaecio — APPROVED** [review]
Suggestion: Support negative numbers to count from the end, e.g. -1 is last, -2 second to last, etc.

File: duplicate_url_discarder/processors/subpath_removal.py (hunk 1 above)
Comment (BurnzZ) [reply duplicate_url_discarder/processors/subpath_removal.py:31]: Good catch! This was from I was casting strings to integers but simply resorted to a stricter int as args.

File: duplicate_url_discarder/processors/subpath_removal.py (hunk 2 above)
Comment (BurnzZ) [reply duplicate_url_discarder/processors/subpath_removal.py:13]: Good catch!

**Review by wRAR — APPROVED** [review]

**Comment by BurnzZ** [comment]
Thanks for the review folks!

**Pull request merged**